Adding instructions for the SLES15 Installation

Dooruk opened the pull request:
Documenting progress 📝 

Also notice the change in the `~/bin/cylc` file and `~./cylc` folder.

Thank you @ashiklom for the SLURM config description!

Dooruk committed:
sles 15 installation instructions

Dooruk committed:
platforms edits

DavidSRussell commented on docs/platforms/discover/configuring_cylc_discover.md:
I also had to enable user execute permissions on `$HOME/bin/cylc`.

DavidSRussell commented on docs/platforms/discover/installing_swell_discover_sles12.md:
I'm guessing it's OK to install in the same directory as previous Swell versions, but maybe you could mention that?

DavidSRussell reviewed (commented):
There are a few things I didn't see mentioned here. To get an experiment to work, I know I had to:
- Use `swell create` option `-p nccs_discover_sles15`
- Make sure user has execute permission on $HOME/bin/cylc
- Change `existing_jedi_build_directory` to `/discover/nobackup/projects/gmao/advda/swell/JediBundles/fv3_soca_SLES15/build-intel-release` in override.yaml
- Change `existing_jedi_source_directory` to `/discover/nobackup/projects/gmao/advda/swell/JediBundles/fv3_soca_SLES15` in override.yaml

and I'm guessing I also had to:
- Change `constraint` to `mil` in $HOME/.swell/swell-slurm.yaml
- Change `platform` to `nccs_discover_sles15` in override.yaml

and I'm guessing `$HOME/.cylc/flow/global.cylc` is not an executable, so I don't need execute permission on that, right?

Dooruk committed:
Merge branch 'develop' into docs/sles15_installation

Dooruk committed:
add description for SLURM directives and clarify create arguments

Dooruk committed:
description for platform specific directives

Dooruk commented:
Thank you so much for your feedback! I added more clarifications on different parts:

>     * Use `swell create` option `-p nccs_discover_sles15`

I added this in the "practical examples" description. 

>     * Make sure user has execute permission on $HOME/bin/cylc
done

>     * Change `existing_jedi_build_directory` to `/discover/nobackup/projects/gmao/advda/swell/JediBundles/fv3_soca_SLES15/build-intel-release` in override.yaml 
>     * Change `existing_jedi_source_directory` to `/discover/nobackup/projects/gmao/advda/swell/JediBundles/fv3_soca_SLES15` in override.yaml

See the  "practical examples" description. We define some of these default folders/builds under  SWELL source code`deployment/platform/{platform}` directories so one should first create a default `experiment.yaml` before accidentally overriding them.

>     * Change `constraint` to `mil` in $HOME/.swell/swell-slurm.yaml

We added `slurm.yaml` file for platform specific entries, see the "SLURM configuration" description.

>     * Change `platform` to `nccs_discover_sles15` in override.yaml
If you run `swell create 3dvar -p nccs_discover_sles15` that is automatically taken care of but you are correct.

> and I'm guessing `$HOME/.cylc/flow/global.cylc` is not an executable, so I don't need execute permission on that, right?

correct

DavidSRussell commented on docs/platforms/discover/installing_swell_discover_sles15.md:
I hit another snag when I realized that my own Swell script was still loading the Spack-Stack modules in `jedi_modules/modules-intel` rather than `jedi_modules/modules-intel-sles15` before loading the Swell module. You mention this in the installation instructions, but maybe it should also be mentioned that these modules should be loaded before every use of Swell? (I don’t know if they’re all needed, but the click module was definitely needed.)

Dooruk replied:
Thanks! Good point, added that.

Dooruk committed:
source module files again
